### PR TITLE
Jira native fixes & more test coverage

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by ppalaga — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

File: integration-tests/jira/src/test/java/org/apache/camel/quarkus/component/jira/it/JiraTest.java [hunk 1]
```diff
@@ -16,33 +16,390 @@
...
+        try {
+            getClient().getIssueClient()
+                    .deleteIssue(issueKey, true)
+                    .claim();
+        } catch (Exception e) {
+            // Ignore - issue may not exist
+        }
+    }
+
+    private static JiraRestClient getClient() {
```
Comment (aldettinger) [inline integration-tests/jira/src/test/java/org/apache/camel/quarkus/component/jira/it/JiraTest.java:389]: I would consider moving this to a @BeforeAll junit 5 step if possible.

**Comment by aldettinger** [comment]
Looks good to me at this stage.

**Comment by jamesnetherton** [comment]
Latest commit addresses review feedback, makes the Jira extension depend on `camel-quarkus-support-spring`, adds another test case and support for running the tests with OAuth creds.

**Review by aldettinger — APPROVED** [review]
Looks good to me.

**Pull request merged**